Add recipe for tab-line-nerd-icons

lucius-martius opened the pull request:
### Brief summary of what the package does

The package defines a global minor mode that applies the icons from nerd-icons to tab-line tabs. This works by adding an advice to `tab-line-tab-name-format-default`. I've decided against defining a custom format function and setting it via `tab-line-tab-name-format-function` because then disabling the global minor mode would be less straight forward. If there is a better way to do this, as with all other aspects, I'm open to suggestions.

### Direct link to the package repository

https://github.com/lucius-martius/tab-line-nerd-icons/tree/master

### Your association with the package

I am the maintainer/author of the package.

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks for this.  Quick first pass -

⸺ `tab-line-nerd-icons.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
- tab-line-nerd-icons.el#L61: It's safer to sharp-quote function names; use `#'`
```

⸺ Package and license (1 files):
- Unreachable package URL in tab-line-nerd-icons.el: https://github.com/lucius_martius/tab-line-nerd-icons

riscy commented:
(By the way, these are relatively minor and the code looks good - should be a quick merge after.)

lucius-martius commented:
@riscy Thanks for your suggestions. I'm not installing the python tool at this point, so for now I've fixed the issues based on your comment.

- I've fixed the typo in the url [here](https://github.com/lucius-martius/tab-line-nerd-icons/commit/16fb68bca1ec705e4a84d67afb3320d4a5a00a46).
- I've fixed the function name that was not sharp-quoted [here](https://github.com/lucius-martius/tab-line-nerd-icons/commit/7a49880f3ae39a8709d6887b26ec84ba2b92360c).
- I've reformatted the commentary section to fit within 80 characters [here](https://github.com/lucius-martius/tab-line-nerd-icons/commit/f63c4dc7c6d6a2ce706c40f1c0c303b1542edc6b).

riscy commented:
Thank you!

riscy commented:
As one last reminder: following these changes, ensure GitHub release 0.1 is up-to-date with your current code and `Package-Version`

lucius-martius commented:
I've force-pushed the tag to the current version including the fixes.

Thanks for merging my recipe.
